resources/documentation/learn_ansible.rst

```diff
@@ -691,5 +691,395 @@ Now regarding to Jinja2:
+
+  - name: My message module
+    debug:
+      msg: "Hello world ! I am the hosts {{ inventory_hostname }}"
+
+Registers
+^^^^^^^^^
+
+Create a file /tmp/valkyrie, and add the string "lenneth" inside, using:
+
+.. code-block:: text
+
+  echo "lenneth" > /tmp/valkyrie
+
+Now, lets use a register to gather this content. Registers are a way to gather data about executed modules. Input data, output data, etc.
+One of the main usages is registers combined with shell or commands, to gather very specific data from the system.
+
+Try:
+
+.. code-block:: yaml
+
+  - name: Get output from a shell command
+    shell: "cat /tmp/valkyrie"
+    register: my_result
+    ignore_errors: True
+
+  - name: Display content of the register
+    debug:
+      msg: "{{ my_result }}"
+
+And when running, output is:
+
+.. code-block:: text
+
+  TASK [debug : Get output from a shell command] **********************************************************************************************************************************************************************************************
+  changed: [management1]
+
+  TASK [debug : My message module] ************************************************************************************************************************************************************************************************************
+  ok: [management1] =>
+    msg:
+      changed: true
+      cmd: cat /tmp/valkyrie
+      delta: '0:00:00.016261'
+      end: '2020-04-17 14:46:04.490684'
+      failed: false
+      rc: 0
+      start: '2020-04-17 14:46:04.474423'
+      stderr: ''
+      stderr_lines: []
+      stdout: lenneth
+      stdout_lines:
+      - lenneth
+
+So, you can use the register to get status (changed: true, so something happens), to get return code (rc at 0 here), stderr and stdout, if it failed, etc.
+
+For example:
+
+.. code-block:: yaml
+
+  - name: My message module
+    debug:
+      msg: "{{ my_result.stdout }}"
+
+Will display "lenneth".
+
+We are going to uses register and debug module to learn generic tasks actions.
+
+Loops
+^^^^^
+
+It is possible to make modules iterate. All possibilities are available here in `loop documentation <https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html>`_ .
+
+BlueBanquise rely on two methods: *with_items* and *loop*. *with_items* is considered replaced by the Ansible team, but since *loop*, the replacement, does not cover all the abilities of *with_items*, it is still widely used.
+
+First example is with loop:
+
+.. code-block:: yaml
+
+  - name: My message module
+    debug:
+      msg: "Values: {{ item }}"
+    loop:
+       - Blue
+       - Red
+       - Green
+       - Yellow
+
+This will execute this module 4 times, with each time an element of the list gives to loop. When using loops in ansible, the variable **item** store the value of the current loop index.
+
+It is also possible to provide loop a list from the inventory, like the one we made before:
```

Comment:
... provide the loop with a list ... like the one we wrote before

Reply:
ok